packages/Interface/src/DTK_C_API.h

```diff
@@ -612,22 +612,41 @@ typedef void ( *DTK_PushFieldDataFunction )( void *user_data,
 /** \brief Prototype function to evaluate a field at a given set of points in a
  *         given set of objects.
  *
- *  Register with a user application using DTK_setUserFunction() by passing
- *  DTK_EVALUATE_FIELD_FUNCTION as the \p type argument.
+ *  This function gives users the ability to use their own interpolant with a
+ *  DTK transfer operator. This function provides a set of coordinates at
+ *  which the function should be evaluated and for each point the local id of
+ *  the object in which the field is located or to which it is nearest. The
+ *  user then interpolates the field onto this point and returns the result.
+ *
+ *  \note Register with a user application using DTK_setUserFunction() by
+ *  passing DTK_EVALUATE_FIELD_FUNCTION as the \p type argument.
  *
  *  \param[in] user_data Custom user data.
- *  \param[in] field_name Name of the field to evaluate.
+ *
+ *  \param[in] field_name Name of the field to evaluate. The user
+ *             implementation of this function should evaluate the field
+ *             associated with this name.
+ *
+ *  \param[in] num_points The number of points at which to evaluate the field.
+ *
+ *  \param[in] space_dim The spatial dimension of the evaluation points.
```

Comment:
Does it make sense to repeat `space_dim` here?  The user specified it already in the `XXXSizeFunction` callback function he registered and it's not like we could pass a different value for this dimension.  I feel it's inconsistent with how we treat `field_dimension`.

Reply:
I would say that the space dimension is not given to the application evaluating the field although I agree that essentially it should be implied by the geometric dimension of the problem.